overlord/snapstate/component.go

```diff
@@ -168,11 +168,26 @@ func doInstallComponent(st *state.State, snapst *SnapState, compSetup *Component
 		}
 	}
 
+	compInstalled := snapst.IsComponentInCurrentSeq(compSi.Component)
+
+	if compSetup.CompType == snap.KernelModulesComponent {
+		// Remove units from previous component
+		if compInstalled {
+			undoSetupKernMod := st.NewTask("cleanup-kernel-modules-component",
+				fmt.Sprintf(i18n.G("Clean-up previous kernel-modules component"),
+					compSi.Component, revisionStr))
+			addTask(undoSetupKernMod)
```

Comment:
what happens if we boot in between here? shouldn't we do a remount instead of removing/cleaning up stuff?